sqlmesh/core/snapshot/definition.py

```diff
@@ -553,6 +553,14 @@ def hydrate_with_intervals_by_identifier(
 
         return result
 
+    @staticmethod
+    def has_drop_alteration(alter_expressions: t.List[exp.AlterTable]) -> bool:
```

Comment:
Why is this a part of the snapshot definition? I don't think this belongs here

Reply:
I wasn't sure where to put it

Reply:
There's nothing specific to snapshot in this method. Shouldn't this be a part of the `SchemaDiffer`?